### ISSUE TITLE
Ubuntu 18.04 detached VM nic

### ISSUE THREAD

**Issue by nickkadams** [terraform-providers/terraform-provider-vsphere#576]
## Terraform and Provider Version
```
Terraform version: 0.11.7
+ provider.vsphere v1.6.0
vSphere 6.7
```
## Debug Output
```
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Looking for post-clone device changes
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Network devices located: ethernet-0
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Current resource set from configuration: (key 0 at <new device>)
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Layout from source: 1 devices over a 1 unit range
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Reading existing devices
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] network_interface.0 (key 4000 at pci:0:7): Reading state
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] FindVirtualDevice: Looking for device with key 4000
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] FindVirtualDevice: Device found: ethernet-0
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] network_interface.0 (key 4000 at pci:0:7): Read finished (key and device address may have changed)
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] network_interface.0 (key 4000 at pci:0:7): Beginning update
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] FindVirtualDevice: Looking for device with key 4000
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] FindVirtualDevice: Device found: ethernet-0
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] network_interface.0 (key 4000 at pci:0:7): Device config operations from update: (edit: *types.VirtualVmxnet3 at key 4000)
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] network_interface.0 (key 4000 at pci:0:7): Update complete
2018-07-28T09:04:11.788-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] applyDeviceChange: Applying changes: (edit: *types.VirtualVmxnet3 at key 4000)
2018-07-28T09:04:11.789-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] applyDeviceChange: Device list before changes: ide-200,ide-201,ps2-300,pci-100,sio-400,keyboard-600,pointing-700,video-500,vmci-12000,lsilogic-1000,cdrom-3000,disk-1000-0,ethernet-0,disk-1000-1
2018-07-28T09:04:11.789-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] applyDeviceChange: Device list after changes: ide-200,ide-201,ps2-300,pci-100,sio-400,keyboard-600,pointing-700,video-500,vmci-12000,lsilogic-1000,cdrom-3000,disk-1000-0,ethernet-0,disk-1000-1
2018-07-28T09:04:11.789-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Post-clone final resource list: (key 4000 at pci:0:7)
2018-07-28T09:04:11.789-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Device list at end of operation: ide-200,ide-201,ps2-300,pci-100,sio-400,keyboard-600,pointing-700,video-500,vmci-12000,lsilogic-1000,cdrom-3000,disk-1000-0,ethernet-0,disk-1000-1
2018-07-28T09:04:11.789-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Device config operations from post-clone: (edit: *types.VirtualVmxnet3 at key 4000)
2018-07-28T09:04:11.789-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:11 [DEBUG] NetworkInterfacePostCloneOperation: Operation complete, returning updated spec
2018-07-28T09:04:16.110-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:16 [DEBUG] Looking for OS family for guest ID "ubuntu64Guest"
2018-07-28T09:04:16.197-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:16 [DEBUG] OSFamily: family for "ubuntu64Guest" is "linuxGuest"
2018-07-28T09:04:16.197-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:16 [DEBUG] Sending customization spec to virtual machine "/us-east-1/vm/awx"
2018-07-28T09:04:17.181-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:17 [DEBUG] Powering on virtual machine "/us-east-1/vm/awx"
2018-07-28T09:04:17.537-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.6.0_x4: 2018/07/28 09:04:17 [DEBUG] vsphere_virtual_machine (ID = 4228e814-93ff-46e2-c8b6-0b554c9d21c7): Waiting for VM customization to complete
vsphere_virtual_machine.vm: Still creating... (20s elapsed)
vsphere_virtual_machine.vm: Still creating... (30s elapsed)
2018/07/28 09:04:37 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:
```
## Expected Behavior
Create Ubuntu 18.04.x VM

## Actual Behavior
Crashes after 30 seconds with above error and a detached vmnic. Same Ubuntu 18.04.1 template can be cloned fine through the VMware UI. 

## Steps to Reproduce
1. Build Ubuntu 18.04.x VM template (packer or manual)

https://github.com/nickkadams/packer-vsphere/tree/master/ubuntu/18

2. terraform apply

https://github.com/nickkadams/terraform-awx

modified of course for Ubuntu 18 config

## Additional Context
Switched from netplan.io to ifupdown legacy config as alternative. Same result. Running latest open-vm-tools 10.2.0 and OS patches, but Ubuntu 18.04 does not work either. Same Terraform works fine with Ubuntu 16.04.

**Comment by bill-rich** [comment]
Hi @nickkadams! It looks like the end of the debug output got cut off. Can you include the error after the last line? Thanks!

**Comment by nickkadams** [comment]
Sorry about that @bill-rich
```
2018/08/07 04:46:21 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* vsphere_virtual_machine.vm:
Virtual machine customization failed on "/us-east-1/vm/awx":

An error occurred while customizing VM awx. For details reference the log file <No Log> in the guest OS.

The virtual machine has not been deleted to assist with troubleshooting. If
corrective steps are taken without modifying the "customize" block of the
resource configuration, the resource will need to be tainted before trying
again. For more information on how to do this, see the following page:
https://www.terraform.io/docs/commands/taint.html
2018/08/07 04:46:21 [TRACE] [walkApply] Exiting eval tree: vsphere_virtual_machine.vm

2018/08/07 04:46:21 [DEBUG] plugin: waiting for all plugin processes to complete...
Error: Error applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.vm: 1 error(s) occurred:

* vsphere_virtual_machine.vm:
Virtual machine customization failed on "/us-east-1/vm/awx":

An error occurred while customizing VM awx. For details reference the log file <No Log> in the guest OS.

The virtual machine has not been deleted to assist with troubleshooting. If
corrective steps are taken without modifying the "customize" block of the
resource configuration, the resource will need to be tainted before trying
again. For more information on how to do this, see the following page:
https://www.terraform.io/docs/commands/taint.html

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by nickkadams** [comment]
There appears to be a correlation to the installed version of open-vm-tools. I remembered I had to version pin in my packer for Ubuntu 16.04. I reran packer with the latest open-vm-tools and terraform fails exactly like Ubuntu 18.04 with the same error after 30 seconds. **New VM from This Template** continues to work through the vCenter UI despite open-vm-tools version. 
```
# working
open-vm-tools 2:10.0.7-3227872-2ubuntu1

# terraform error
open-vm-tools 2:10.2.0-3~ubuntu0.16.04.1
```

**Comment by rismoney** [comment]
It is strange, because I customized a new node cleanly yesterday with 2.10.2 and today I couldn't.  Nothing changed in source between my 2 attempts. I switched to xenial source and switched vmtools and it worked as you reported.  I am using vsphere 6, fwiw.  I got error however doing cust thru vsphere too, imolying for me it wasn't TF specific...

**Comment by jstoja** [comment]
@nickkadams Could you share with us how you provisioned the template you're using (+ the ovf file to check the network interfaces template)?
Have you checked the logs of the task marked as error in vCenter? If yes could you paste them here?

**Comment by jstoja** [comment]
I confirm that I manage to reproduce it. I used the exact same packer build file (just changed the ISO), and same terraform configuration. The VM correctly provisions and customizes from the template with Ubuntu 16.04, but I get the same error as @nickkadams when using 18.04.

**Comment by jstoja** [comment]
Could it be linked to https://github.com/vmware/open-vm-tools/issues/240 ? It might be that the migration from ifupdown to netplan broke some things here.

**Comment by jstoja** [comment]
@bill-rich I guess this can be closed since it's apparently not related to Terraform.

**Comment by nickkadams** [comment]
@jstoja while I agree that is largely tied to open-vm-tools versioning, why would the vCenter UI be able to clone the exact same Ubuntu 16 and 18 templates that were referenced in my packer repo. Terraform fails where the UI succeeds. Additionally, I see the same behavior on Ubuntu 16.04 with 10.2.

**Comment by nickkadams** [comment]
@jstoja is your template on open-vm-tools using 10.2 or 10.0? 

https://github.com/nickkadams/packer-vsphere/blob/master/ubuntu/16/ubuntu16.json#L82

I version pin to make 16.04 work.

**Comment by jstoja** [comment]
16.04: `2:10.2.0~3ubuntu0.16.04.1`
18.04 (working with cloud-init): `2:10.2.0-3ubuntu3`

Something that I also had to do is to power up and down the vm template so vmware is aware that open-vm-tools is correctly installed on it (if not I get another error than the on you have so not related).

**Comment by pryorda** [comment]
It appeared to just be a symlink issue for me

https://github.com/pryorda/ubuntuPackerImage/commit/3cc8a51c6a3d6077f704f73789bfb4c00eb43512#diff-9cf3804da628e5c7ac06324909edbf4dR70

**Comment by nickkadams** [comment]
@pryorda thanks for the suggestion, but I had previously tried that and again today with no success. Latest 16.04 open-vm-tools does not work from my TF. Version pinning has been my workaround with no fix for 18.04.

**Comment by nickkadams** [comment]
https://kb.vmware.com/s/article/56409
and
https://github.com/vmware/open-vm-tools/issues/240
reference the fixes for Ubuntu 18.04. closing as it is not a TF bug.
